sdk/tables/azure-data-tables/azure/data/tables/_encoder.py

```diff
@@ -0,0 +1,112 @@
+import abc, enum
+from typing import Any, Optional, Tuple, Union, Mapping, TypeVar, Generic, Dict
+from uuid import UUID
+from datetime import datetime
+from math import isnan
+
+from ._serialize import EdmType
+from ._entity import TableEntity
+from ._deserialize import _convert_to_entity
+from ._common_conversion import _encode_base64, _to_utc_datetime
+
+_ODATA_SUFFIX = "@odata.type"
+T = TypeVar("T")
+
+
+class TableEntityEncoderABC(abc.ABC, Generic[T]):
+    def prepare_key(self, key: str) -> str:
+        """Duplicate the single quote char to escape."""
+        try:
+            return key.replace("'", "''")
+        except (AttributeError, TypeError):
+            raise TypeError('PartitionKey or RowKey must be of type string.')
+
+    def prepare_value(self, name: Optional[str], value: Any) -> Tuple[Optional[EdmType], Optional[Union[str, int, float, bool]]]:
+        """This is a utility to allow customers to reproduce our internal encoding for types like datetime, bytes, float and int64.
```

Comment:
Could we update all the class and method docstrings to be customer friendly?
Currently they are all our internal notes/discussions.

Reply:
Updated all public function docstring in this file.